rotkehlchen/tests/unit/test_price_historian.py

```diff
@@ -336,3 +362,41 @@ def test_price_priority_order():
     """Test to ensure that we detect changes on the constant value returned"""
     _, priority_value = _prioritize_manual_balances_query()
     assert priority_value == HistoricalPriceOracle.MANUAL.serialize_for_db()
+
+
+@pytest.mark.vcr
+@pytest.mark.parametrize('mocked_price_queries', [mocked_prices])
+@pytest.mark.parametrize('ethereum_manager_connect_at_start', [(INFURA_ETH_NODE,)])
+def test_uniswap_v2_position_history(price_historian: PriceHistorian):
+    price = price_historian.query_uniswap_position_price(
+        pool_token=EvmToken.initialize(
+            address=string_to_evm_address('0xB4e16d0168e52d35CaCD2c6185b44281Ec28C9Dc'),
+            chain_id=ChainID.ETHEREUM,
+            token_kind=EvmTokenKind.ERC20,
+            protocol=UNISWAP_PROTOCOL,
+            decimals=18,
+        ),
+        pool_token_amount=FVal('0.015374510179577256'),
+        timestamp=Timestamp(1742814047),
+    )
+
+    assert price == Price(FVal('3599499.14614648204764245778042706722040502098628449234280700474399384952003946'))  # noqa: E501
+
+
+@pytest.mark.vcr
+@pytest.mark.parametrize('mocked_price_queries', [mocked_prices])
+@pytest.mark.parametrize('ethereum_manager_connect_at_start', [(INFURA_ETH_NODE,)])
+def test_uniswap_v3_position_history(price_historian: PriceHistorian):
+    price = price_historian.query_uniswap_position_price(
+        pool_token=EvmToken.initialize(
```

Comment:
this is what I'm not sure how it works. Check that you call `query_uniswap_position_price` but the logic for processing the PnL uses the function price so I understand that something is missing to make your changes work with the app. Remember that the end goal is to solve a problem in the app

Reply:
What was the answer here? @prodesert22 I need to see an answer to **every single** comment. not just resolving them without an answer

Reply:
this was solved calling the logic in the the priceHistorian. The place where you asked about using kwargs